src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -3706,6 +3711,325 @@ def aks_check_network_outbound(
+        )
+        return result
+    except Exception as ex:
+        logger.error("Failed to get K8s extension.\nError: %s", ex)
+
+
+def list_k8s_extension_types(
+    cmd,
+    client,
+    location=None,
+    resource_group_name=None,
+    cluster_name=None,
+    release_train=None
+):
+    k8s_extension_custom_mod = get_k8s_extension_module(CONST_K8S_EXTENSION_CUSTOM_MOD_NAME)
+    client_factory = get_k8s_extension_module(CONST_K8S_EXTENSION_CLIENT_FACTORY_MOD_NAME)
+    client = client_factory.cf_k8s_extension_types(cmd.cli_ctx)
+    try:
+        if location:
+            result = k8s_extension_custom_mod.list_extension_types_by_location(
+                client,
+                location,
+            )
+            return result
+        if cluster_name and resource_group_name:
+            result = k8s_extension_custom_mod.list_extension_types_by_cluster(
+                client,
+                resource_group_name,
+                cluster_name,
+                "managedClusters",
+                release_train=release_train,
+            )
+            return result
+    except Exception as ex:
+        logger.error("Failed to list K8s extension types by location.\nError: %s", ex)
+
+
+# get K8s extension type
+def show_k8s_extension_type(
+    cmd,
+    client,
+    extension_type,
+    location=None,
+    resource_group_name=None,
+    cluster_name=None
+):
+    if not check_if_extension_type_is_in_allow_list(extension_type.lower()):
+        raise ValidationError(f"Failed to get extension type {extension_type.lower()} by location " +
+                              f"as it is not in allowed list of extension types {allowedListOfExtensions}")
+    k8s_extension_custom_mod = get_k8s_extension_module(CONST_K8S_EXTENSION_CUSTOM_MOD_NAME)
+    client_factory = get_k8s_extension_module(CONST_K8S_EXTENSION_CLIENT_FACTORY_MOD_NAME)
+    client = client_factory.cf_k8s_extension_types(cmd.cli_ctx)
+    try:
+        if location:
+            result = k8s_extension_custom_mod.show_extension_type_by_location(
+                client,
+                location,
+                extension_type=extension_type,
+            )
+            return result
+        if cluster_name and resource_group_name:
+            result = k8s_extension_custom_mod.show_extension_type_by_cluster(
+                client,
+                resource_group_name,
+                cluster_name,
+                "managedClusters",
+                extension_type,
+            )
+            return result
+    except Exception as ex:
+        logger.error("Failed to get K8s extension types by location.\nError: %s", ex)
+
+
+# list version by location
+def list_k8s_extension_type_versions(
+    cmd,
+    client,
+    extension_type,
+    location=None,
+    resource_group_name=None,
+    cluster_name=None,
+    release_train=None,
+    major_version=None,
+    show_latest=False
+):
+    if not check_if_extension_type_is_in_allow_list(extension_type.lower()):
```

Comment:
minor: you repeat this `if not check: raise` pattern quite a bit.

Might be cleaner to just turn `check_if_extension_type_is_in_allow_list` into `raise_if_extension_type_not_allowed`, and put the `raise ValidationError` + error text in the helper too, so you can just call:

`raise_if_extension_type_not_allowed(extension_type, allowed_extensions)`